lib/kitchen/directions/bake_numbered_exercise/v1.rb

```diff
@@ -2,15 +2,17 @@
 
 module Kitchen::Directions::BakeNumberedExercise
   class V1
-    def bake(exercise:, number:)
+    def bake(exercise:, number:, suppress_solution: false)
       problem = exercise.problem
       solution = exercise.solution
 
       problem_number = "<span class='os-number'>#{number}</span>"
 
-      if solution.present?
+      if solution.present? && !suppress_solution
         problem_number = "<a class='os-number' href='##{exercise.id}-solution'>#{number}</a>"
         bake_solution(exercise: exercise, number: number)
+      elsif suppress_solution
+        solution&.trash
```

Comment:
To not have to check `suppress_solution` twice, you can:

```ruby
if solution.present? 
  if suppress_solution
    solution.trash
  else
    problem_number = "<a class='os-number' href='##{exercise.id}-solution'>#{number}</a>"
    bake_solution(exercise: exercise, number: number)
  end
end
```

Also since you know that `solution.present?` in this way you don't need the `&` safe navigation operator.  

These two changes make this code reads a little more clearly to me.  Fewer checks.

Reply:
Also I realize this reverses Kendra's comment. :-)

Reply:
okay this is probably better... my bad natalia

Reply:
No problem :wink: